PolyEngine/Engine/Src/Entity.hpp

```diff
@@ -48,13 +47,7 @@ namespace Poly
 		/// @return A pointer to a component or nullptr if it does not exist.
 		//////////////////////////////
 		template<class T>
-		T* GetComponent()
-		{
-			if (HasComponent(GET_COMPONENT_ID(EntityWorld->GetEngine(), T)))
-				return static_cast<T*>(Components[GET_COMPONENT_ID(EntityWorld->GetEngine(), T)]);
-			else
-				return nullptr;
-		}
+		T* GetComponent(); //defined in World.hpp due to circular inclusion problem
```

Comment:
This is bad resolution of this problem. Also i don't see the circular inclussion. please elaborate.